wait/log.go

```diff
@@ -24,21 +24,27 @@ type LogStrategy struct {
 // NewLogStrategy constructs with polling interval of 100 milliseconds and startup timeout of 60 seconds by default
 func NewLogStrategy(log string) *LogStrategy {
 	return &LogStrategy{
-		startupTimeout: defaultStartupTimeout(),
-		Log:            log,
-		Occurrence:     1,
-		PollInterval:   defaultPollInterval(),
+		timeout:      defaultTimeout(),
+		Log:          log,
+		Occurrence:   1,
+		PollInterval: defaultPollInterval(),
 	}
-
 }
 
 // fluent builders for each property
 // since go has neither covariance nor generics, the return type must be the type of the concrete implementation
-// this is true for all properties, even the "shared" ones like startupTimeout
+// this is true for all properties, even the "shared" ones like timeout
 
 // WithStartupTimeout can be used to change the default startup timeout
-func (ws *LogStrategy) WithStartupTimeout(startupTimeout time.Duration) *LogStrategy {
-	ws.startupTimeout = startupTimeout
+//
+// Deprecated: use WithTimeout instead
+func (s *LogStrategy) WithStartupTimeout(timeout time.Duration) *LogStrategy {
+	return s.WithTimeout(timeout)
```

Comment:
Not sure how Go's best practices handle deprecation patterns: do you think we can log the message about the new method?

Reply:
I'm interested in how to notify users and consumer code about deprecations. In Java you have `@deprecated` annotation to warn the compiler about it, and for Go I've understood that it's done as you suggested: https://github.com/golang/go/wiki/Deprecated

Will resolve this conversation, and will wait for @gianarb for the merge.

Thanks!